daemon/container_operations_unix.go

```diff
@@ -120,19 +122,21 @@ func (daemon *Daemon) setupIpcDirs(c *container.Container) error {
 				return err
 			}
 
-			shmSize := int64(daemon.configStore.ShmSize)
-			if c.HostConfig.ShmSize != 0 {
-				shmSize = c.HostConfig.ShmSize
-			}
-			shmproperty := "mode=1777,size=" + strconv.FormatInt(shmSize, 10)
+			shmproperty := "mode=1777,size=" + strconv.FormatInt(c.HostConfig.ShmSize, 10)
 			if err := unix.Mount("shm", shmPath, "tmpfs", uintptr(unix.MS_NOEXEC|unix.MS_NOSUID|unix.MS_NODEV), label.FormatMountLabel(shmproperty, c.GetMountLabel())); err != nil {
 				return fmt.Errorf("mounting shm tmpfs: %s", err)
 			}
 			if err := os.Chown(shmPath, rootIDs.UID, rootIDs.GID); err != nil {
 				return err
 			}
+			c.ShmPath = shmPath
 		}
 
+	case ipcMode.IsPrivate():
+		c.ShmPath = ""
+
+	default:
```

Comment:
I am asking because, even with the new daemon and the default-ipc-mode set to 'private' the original problem (of container's ability to correctly checkpoint/restore) won't be fixed because of such pre-existing containers for which we apply the old default.

I'm not quite sure how to address that. One way is to add `--ipc` to `docker update` (with a requirement for a container to be stopped). Another way is to enforce the current daemon default (instead of falling back to "shareable" the way it is now), but it might break backward compatibility.

Any thoughts?